packages/iris-grid/src/IrisGrid.tsx

```diff
@@ -2330,9 +2341,23 @@ export class IrisGrid extends Component<IrisGridProps, IrisGridState> {
     );
   }
 
-  toggleGotoRow(): void {
+  toggleGotoRow(row: string): void {
     const { isGotoRowShown } = this.state;
-    this.setState({ isGotoRowShown: !isGotoRowShown });
+    if (!isGotoRowShown) {
+      this.setState({
+        isGotoRowShown: !isGotoRowShown,
```

Comment:
This is kind of confusing... in both cases you're setting the same values, except for `selectGotoRowInput`. You could write it just as one setState:
```
this.setState({
  isGotoRowShown: !isGotoRowShown,
  gotoRow: row,
  gotoRowError: '',
  selectGotoRowInput: !isGotoRowShown,
});
```

Reply:
I modified behavior slightly. After playing with it a bit, I actually preferred it to stay open on subsequent context menus calls instead of toggle, but still update the row value with the row it was invoked from.

I also removed the need for `selectGotoRowInput` by just checking document.activeElement to determine if it needs to be selected.